test/LockedNORI.test.ts

```diff
@@ -0,0 +1,1166 @@
+        await expect(pausableFunction({ lNori, hre })).revertedWith(
+          'Pausable: paused'
+        );
+      });
+    }
+
+    it(`will not allow tokens to be deposited when the contract is paused`, async () => {
+      const { lNori, bpNori, grant, hre } = await setupWithGrant();
+      const { namedAccounts, namedSigners, ethers } = hre;
+
+      await expect(lNori.connect(namedSigners.admin).pause()).to.emit(
+        lNori,
+        'Paused'
+      );
+
+      await expect(
+        lNori.depositFor(namedAccounts.supplier, formatTokenAmount(1))
+      ).revertedWith('Pausable: paused');
+    });
+  });
+
+  it(`will not allow tokens to be withdrawn when the contract is paused`, async () => {
+    const { lNori, bpNori, grant, hre } = await setupWithGrant();
+    const { namedAccounts, namedSigners, ethers } = hre;
+
+    await advanceTime({ hre, timestamp: grant.unlockEndTime });
+
+    await expect(lNori.connect(namedSigners.admin).pause()).to.emit(
+      lNori,
+      'Paused'
+    );
+
+    await expect(
+      lNori
+        .connect(namedSigners.supplier)
+        .withdrawTo(namedAccounts.supplier, formatTokenAmount(1))
+    ).revertedWith('Pausable: paused');
+  });
+});
+
+describe('initialization', () => {
+  // it.todo('should fire events');
+  describe('roles', () => {
+    for (const { role } of [
+      { role: 'DEFAULT_ADMIN_ROLE' },
+      { role: 'PAUSER_ROLE' },
+      { role: 'TOKEN_GRANTER_ROLE' },
+    ] as const) {
+      it(`will assign the role ${role} to the deployer and set the DEFAULT_ADMIN_ROLE as the role admin`, async () => {
+        const { lNori, hre } = await setupTest();
+        expect(
+          await lNori.hasRole(await lNori[role](), hre.namedAccounts.admin)
+        ).to.be.true;
+        expect(await lNori.getRoleAdmin(await lNori[role]())).to.eq(
+          await lNori.DEFAULT_ADMIN_ROLE()
+        );
+        expect(await lNori.getRoleMemberCount(await lNori[role]())).to.eq(1);
+      });
+    }
+  });
+});
+
+describe('role access', () => {
+  // it.todo('only the token granter role can deposit tokens')
+  describe('roles', () => {
+    describe('TOKEN_GRANTER_ROLE', () => {
+      for (const { role, accountWithRole, accountWithoutRole } of [
+        {
+          role: 'TOKEN_GRANTER_ROLE',
+          accountWithRole: 'admin',
+          accountWithoutRole: 'investor1',
+        } as const,
+      ]) {
+        it(`accounts with the role "${role}" can use "batchRevokeUnvestedTokenAmounts" whilst accounts without the role "${role}" cannot`, async () => {
+          const { lNori, grantAmount, grant, hre } = await setupWithGrant(
+            (parameters) => employeeParameters(parameters)
+          );
+          const { namedAccounts, namedSigners } = hre;
+          const roleId = await lNori[role]();
+          expect(await lNori.hasRole(roleId, namedAccounts[accountWithRole])).to
+            .be.true;
+          expect(
+            await lNori
```

Comment:
[await order is important for this](https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#emitting-events)
```suggestion
          await expect(
            lNori
```